docs/index.rst

```diff
@@ -29,20 +29,333 @@ Asset filters are added in the same way as any other filters.
+  - **asset_name** - Either a literal asset name to match or a regular expression.
 
-  - **remove**: This action will be passed a datapoint name as an argument or a datapoint type. A datapoint with that name will be removed from the asset as it passed through the asset filter. If a type is passed then all data points of that type will be removed.
+  - **action** - The action the rule will take.
 
-    .. list-table:: Supported data types for "remove" action
+The sections below document the details of each of the different rules and the parameters supported by those actions.
+
+Include Rule
+############
+
+This rule takes no extra parameters, if the asset name matches that of the reading then the reading will be included in the output.
+
+.. code-block:: JSON
+
+   {
+       "asset_name" : "pump417",
+       "action"     : "include"
+   }
+
+Exclude Rule
+############
+
+This rule takes no extra parameters, if the asset name matches that of the reading then the reading will be excluded in the output.
+
+.. code-block:: JSON
+
+   {
+       "asset_name" : "pump417",
+       "action"     : "exclude"
+   }
+
+Rename Rule
+###########
+
+The rename rule takes an extra parameter which is the new name of the asset that will be used for any matching reading.
+
+.. code-block:: JSON
+
+   {
+       "asset_name"     : "pump417",
+       "action"         : "exclude",
+       "new_asset_name" : "SlurryPump417"
+   }
+
+If the *asset_name* property is a regular expression then the rule can be used to rename several assets in a single rule. In this case elements of the matched name can be used in the new name using standard regular expression syntax.
+
+.. code-block:: JSON
+
+   {
+       "asset_name"     : "pump([0-9]*)",
+       "action"         : "exclude",
+       "new_asset_name" : "SlurryPump$1"
+   }
+
+In the above example any asset called pump with a numeric suffix will be matched, the new name for the asset will be SlurryPump with the same numeric suffix added. The *$1* in the new asset name will be substituted with the matching text of the first bracketed expression in the regular expression given in *asset_name*.
+
+Remove Rule
+###########
+
+The remove rule will remove one or more datapoints from a reading. The remove rule must be passed one of a number of parameters that are used to define the datapoints to be removed. These are
+
+  - **datapoint**: Remove datapoints that match the string value of the datapoint property. The value may be a literal datapoint name or a regular expression.
+
+  - **datapoints**: remove datapoints that match any of the values given in the array of strings that are the value of the datapoints property. The strings in the array may be a mixture of literal datapoint names or regular expressions.
+
+  - **type**: remove datapoints whose type matches the string value of the type property. 
+
+To remove a single named datapoint
```

Comment:
For this and all of the text below that labels an example, end with a colon, as in "To remove a single named datapoint:"